kms/tpmkms/tpmkms.go

```diff
@@ -326,9 +340,32 @@ func (k *TPMKMS) CreateKey(req *apiv1.CreateKeyRequest) (*apiv1.CreateKeyRespons
 		return nil, fmt.Errorf("failed parsing %q: %w", req.Name, err)
 	}
 
-	v, ok := signatureAlgorithmMapping[req.SignatureAlgorithm]
-	if !ok {
-		return nil, fmt.Errorf("TPMKMS does not support signature algorithm %q", req.SignatureAlgorithm)
+	ctx := context.Background()
+	caps, err := k.tpm.GetCapabilities(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not get TPM capabilities %w", err)
+	}
+
+	var (
+		v  algorithmAttributes
+		ok bool
+	)
```

Comment:
Defining these here seems unnecessary if you make them assignments in both parts below.

Reply:
`v` is used further down, I could refactor more of this code if you feel strongly.